tests/std/tests/P0784R7_library_support_for_more_constexpr_containers/test.cpp

```diff
@@ -272,7 +267,249 @@ constexpr void test_compiletime() {
+        A<int>* a  = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            construct_at(a + i);
+        }
+        destroy(a, a + 10);
+        alloc.deallocate(a, 10);
+    }
+    {
+        auto alloc           = allocator<nontrivial_A<int>>{};
+        nontrivial_A<int>* a = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            construct_at(a + i);
+        }
+        destroy(a, a + 10);
+        alloc.deallocate(a, 10);
+    }
+#ifdef __cpp_lib_concepts
+    {
+        auto alloc = allocator<A<int>>{};
+        A<int>* a  = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            ranges::construct_at(a + i);
+        }
+        ranges::destroy(a, a + 10);
+        alloc.deallocate(a, 10);
+    }
+    {
+        auto alloc           = allocator<nontrivial_A<int>>{};
+        nontrivial_A<int>* a = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            ranges::construct_at(a + i);
+        }
+        ranges::destroy(a, a + 10);
+        alloc.deallocate(a, 10);
+    }
+    {
+        auto alloc = allocator<A<int>>{};
+        A<int>* a  = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            ranges::construct_at(a + i);
+        }
+        std::span s{a, 10};
+        ranges::destroy(s);
+        alloc.deallocate(a, 10);
+    }
+    {
+        auto alloc           = allocator<nontrivial_A<int>>{};
+        nontrivial_A<int>* a = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            ranges::construct_at(a + i);
+        }
+        std::span s{a, 10};
+        ranges::destroy(s);
+        alloc.deallocate(a, 10);
+    }
+#endif // __cpp_lib_concepts
+    {
+        auto alloc = allocator<A<int>>{};
+        A<int>* a  = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            construct_at(a + i);
+        }
+        destroy_n(a, 10);
+        alloc.deallocate(a, 10);
+    }
+    {
+        auto alloc           = allocator<nontrivial_A<int>>{};
+        nontrivial_A<int>* a = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            construct_at(a + i);
+        }
+        destroy_n(a, 10);
+        alloc.deallocate(a, 10);
+    }
+#ifdef __cpp_lib_concepts
+    {
+        auto alloc = std::allocator<A<int>>{};
+        A<int>* a  = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            ranges::construct_at(a + i);
+        }
+        ranges::destroy_n(a, 10);
+        alloc.deallocate(a, 10);
+    }
+    {
+        auto alloc           = std::allocator<nontrivial_A<int>>{};
+        nontrivial_A<int>* a = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            ranges::construct_at(a + i);
+        }
+        ranges::destroy_n(a, 10);
+        alloc.deallocate(a, 10);
+    }
+#endif // __cpp_lib_concepts
+}
+static_assert((test_compiletime_destroy_variants(), true));
+
+template <class T, bool Construct = false, bool Destroy = false>
+struct Alloc {

```

Comment:
This allocator is missing:
* A rebinding constructor
* `operator==` (C++20 will rewrite `operator!=`)
* `rebind` support

For `rebind`, there's a default for allocators of the form `SomeAllocator<T, Args>` (see WG21-N4878 16.4.4.6.1 [allocator.requirements.general]/3), but the `bool` template parameters interfere with this.